miner/worker.go

```diff
@@ -1508,6 +1508,34 @@ func (w *worker) getSealingBlock(params *generateParams) *newPayloadResult {
 	}
 }
 
+func (w *worker) tryWaitProposalDoneWhenStopping() {
+	posa, ok := w.engine.(consensus.PoSA)
+	// if the consensus is not PoSA, just skip waiting
+	if !ok {
+		return
+	}
+
+	currentHeader := w.chain.CurrentBlock()
+	startBlock, endBlock, err := posa.NextProposalBlock(w.chain, w.chain.CurrentBlock(), w.coinbase)
+	if err != nil {
+		log.Warn("Failed to get next proposal block", "err", err)
+		return
+	}
+
+	currentBlock := currentHeader.Number.Uint64()
+	waitSecs := (endBlock - currentBlock) * posa.BlockInterval()
```

Comment:
so, if 19 seconds later, the miner will in turn, will we shutdown the validator imediately?